### PR TITLE
Emit all shellcheck errors using github action toolkit annotations and check for tab indentation

### PR DESCRIPTION
**Pull request by cartercanedy**
Tabs are problematic when rendering previews of shell scripts, as it results in a disparity between the position of expected rendered characters and where they may actually be rendered on the screen, and having mixed space/tab indentation in the same file is generally problematic as well.

## Type of Change
- [x] Bug fix
- [x] Refactoring

## Description
- Emit all errors found within the changed files using gh actions toolkit annotations, before, the workflow would fail on the first shellcheck invocation that had an error emission
- Add a tab indentation check to enforce consistency in code style
- removed all current tab indentations that I've found in shell scripts

## Testing
Verified using workflow runs within my own repository

## Impact
Helps provide complete error information from both shellcheck and tab indentation detection

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/applications-setup/browser-setup.sh [hunk 1]
```diff
@@ -88,38 +88,38 @@ install_firefox() {
...
+            "$ESCALATION_TOOL" zypper ar -ef https://rpm.librewolf.net librewolf
+            "$ESCALATION_TOOL" zypper refresh
+            "$ESCALATION_TOOL" zypper --non-interactive install librewolf
+            ;;
+        pacman)
+            "$AUR_HELPER" -S --needed --noconfirm librewolf-bin
+            ;;
+        *)
+            printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
+            exit 1
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/browser-setup.sh:120]: these formatting changes are going to conflict with #648

File: core/tabs/applications-setup/Developer-tools/sublime-setup.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 . ../../common-script.sh
 
 installSublime() {
-	if ! command_exists sublime; then
+    if ! command_exists sublime; then
```
Comment (nnyyxxxx) [inline core/tabs/applications-setup/Developer-tools/sublime-setup.sh:6]: these formatting changes are going to conflict with #648

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

**Comment by nnyyxxxx** [comment]
also the browser-setup and sublime-setup formatting changes are already implemented in #648 @cartercanedy

File: core/tabs/applications-setup/Developer-tools/sublime-setup.sh (hunk 2 above)
Comment (cartercanedy) [reply core/tabs/applications-setup/Developer-tools/sublime-setup.sh:6]: this should be easy to resolve with a simple rebase, not worried about it

**Comment by cartercanedy** [comment]
> also the browser-setup formatting changes are already implemented in #648 @cartercanedy

I need those implemented here to pass the new shellcheck ci workflow. You should be able to rebase your changes on top of mine

**Comment by cartercanedy** [comment]
@ChrisTitusTech I'd recommend committing this before pulling in any shell scripts to get better shellcheck errors and catch formatting issues

**Review by nnyyxxxx — APPROVED** [review]

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**